flink/v1.14/flink/src/main/java/org/apache/iceberg/flink/FlinkDynamicTableFactory.java

```diff
@@ -142,6 +143,12 @@ public Set<ConfigOption<?>> optionalOptions() {
     Set<ConfigOption<?>> options = Sets.newHashSet();
     options.add(CATALOG_DATABASE);
     options.add(CATALOG_TABLE);
+    options.add(FlinkLookupOptions.LOOKUP_CACHE_MAX_ROWS);
+    options.add(FlinkLookupOptions.LOOKUP_CACHE_TTL);
+    options.add(FlinkLookupOptions.LOOKUP_CACHE_IGNORE_EMPTY);
+    options.add(FlinkLookupOptions.LOOKUP_MAX_RETRIES);
+    options.add(FlinkLookupOptions.LOOKUP_BASE_RETRY_BACKOFF_MILLS);
+    options.add(FlinkLookupOptions.LOOKUP_MAX_RETRY_BACKOFF_MILLS);
```

Comment:
Can we separate caching from the lookup source implementation?

Reply:
sorry, i don't understand what you mean.
FlinkLookupFunction#eval method has no return value, how to cache function return value?

https://github.com/apache/flink/blob/master/flink-connectors/flink-connector-hbase-base/src/main/java/org/apache/flink/connector/hbase/source/HBaseRowDataLookupFunction.java#L96
https://github.com/apache/flink/blob/master/flink-connectors/flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/table/JdbcRowDataLookupFunction.java#L146